src/frontend/src/binder/insert.rs

```diff
@@ -87,6 +89,48 @@ impl Binder {
             .map(|c| c.data_type().clone())
             .collect();
 
+        // Adding Null values in case user did not specify all columns. E.g.
+        // create table t1 (v1 int, v2 int); insert into t1 (v2) values (5);
+        let (source, nulls_inserted) = match source {
+            Query {
+                with,
+                body: SetExpr::Values(values),
+                order_by,
+                limit,
+                offset,
+                fetch,
+            } => {
+                // We only need to insert nulls if user did not define all columns
+                let (new_body, nulls_inserted) = if !values.0.is_empty()
+                    && !values.0.iter().all(|v| v.len() == expected_types.len())
+                {
+                    // all value lists should have same length. Always add same num of nulls
+                    // Illegal statement: insert into t values (1, 2), (3, 4, 5);
```

Comment:
Since this check happens in `bind_values`, would it be better to move the logic of filling NULLs there too? 🤔

Reply:
Got it to work. It feels a bit messy, but I am possibly saving one extra iteration in some cases